File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        
        int res = 0;

        for(int i=0 ; i<s.length() ; i++){

            int string_index = i+1;
            int reversed_index = 26 - ((int) s.charAt(i) % 97);

            res += string_index * reversed_index;

        }

        return res;

    }
}

File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.py
class Solution(object):
    def reverseDegree(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0

        for i in range(len(s)):
            string_index = i+1
            reversed_index = 26 - ( ord(s[i])%97 )
            res += string_index * reversed_index

        return res
        